### DISCUSSION TITLE
Возможность деактивации ссылок

### DISCUSSION THREAD

**Discussion by MadZee-M** [moonshine-software/moonshine#1057] [Ideas]
Доброго времени суток!

Было бы замечательно, если была бы возможность переключать активность ссылок.

![image](https://github.com/moonshine-software/moonshine/assets/32099597/7dc6e53b-d79b-4caa-bc25-8720bd8bc104)

Там создать виде атрибута или даже лучше создать отдельного метода изменения этого свойства.

**Comment by lee-to** [comment]
О чем тут речь?

**Reply by MadZee-M** [reply]
Допустим, я вызываю метод у 
`
Text::make('Status')->link('/success')
`

Но мне ссылка нужно только, когда статус success.
И приходится делать что-то типа 

`
Text::make('Status')->link(fn($value, Field $field) => $field->getData()->status === 'success' ? '/success' : '#')
`
Но все равно я вижу ссылку.

---

А я предлагаю сделать что-то типа
`
Text::make('Status')
  ->canLink(fn(Field $field) => $field->getData()->status === 'success' )
  ->link('/success')
`

И если передана функция в `->link(fn() => ...)`, то она не должна вызываться, когда `->canLink(false)`

**Reply by lee-to** [reply]
Метод линк же работает в форме где у нас есть getItem, почему не сделать через ->when ?

**Reply by MadZee-M** [reply]
Через ->when узнал после того как написал.
Но в любом случае, думаю было бы хорошо, если был бы еще и подобный метод ->canLink.
Спасибо за обратную связь!
